File: solutions/java/src/librarymanagementsystem/Member.java
package librarymanagementsystem;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.UUID;

public class Member {
    private final String id;
    private final String name;
    private final String contactInfo;
    private final List<Loan> loans;

    public Member(String name, String contactInfo) {
        this.id = UUID.randomUUID().toString();
        this.name = name;
        this.contactInfo = contactInfo;
        this.loans = Collections.synchronizedList(new ArrayList<>());
    }

    public int getBorrowedCount() { return loans.size(); }

    public synchronized void addLoan(Loan loan) {
        loans.add(loan);
    }

    public synchronized void removeLoan(Loan loan) {
        loans.remove(loan);
    }

    public List<Loan> getLoans() {
        return loans;
    }

    public String getId() {
        return id;
    }

    public String getName() {
        return name;
    }
}

File: solutions/python/librarymanagementsystem/member.py
from typing import List
from loan import Loan
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from library_item import LibraryItem
    from loan import Loan

class Member:
    def __init__(self, member_id: str, name: str):
        self.id = member_id
        self.name = name
        self.loans: List['Loan'] = []

    def update(self, item: 'LibraryItem') -> None:
        """Observer update method"""
        print(f"NOTIFICATION for {self.name}: The book '{item.get_title()}' you placed a hold on is now available!")

    def add_loan(self, loan: 'Loan') -> None:
        self.loans.append(loan)

    def remove_loan(self, loan: 'Loan') -> None:
        if loan in self.loans:
            self.loans.remove(loan)

    def get_id(self) -> str:
        return self.id

    def get_name(self) -> str:
        return self.name

    def get_loans(self) -> List['Loan']:
        return self.loans